### PR TITLE
Added support for SparkFun SAMD21 Dev and Mini Breakout boards

### PR DESCRIPTION
**Pull request by ShawnHymel**
...and updated test_client.py to check for equality on bp.ignore.

### CODE REVIEW

File: mu/modes/adafruit.py [hunk 1]
```diff
@@ -64,6 +64,10 @@ class AdafruitMode(MicroPythonMode):
         (0x239A, 0x803C),  # future board
         (0x239A, 0x803E),  # future board
         (0x239A, None),    # Any Adafruit Boards
+
+        # Non-Adafruit boards
+        (0x1B4F, 0x8D22),  # SparkFun SAMD21 Mini Breakout
+        (0x1B4F, 0x8D23),  # SparkFun SAMD21 Dev Breakout
```
Comment (carlosperate) [inline mu/modes/adafruit.py]: Is this Vendor ID specific to Sparkfun boards?
(i.e. although probably not applicable for this case, some boards use the same type of UART-to-USB bridges, so they could end up with the same VID/PID).

Comment (ShawnHymel) [reply mu/modes/adafruit.py]: Yes, the VID is unique to SparkFun. I had @jimblom from SparkFun pull the PIDs for the boards specifically for CircuitPython.

Comment (carlosperate) [reply mu/modes/adafruit.py]: Great, thanks @ShawnHymel.

**Comment by ntoll** [comment]
@ShawnHymel all merged. Thank you very much for your contribution..! BTW, since there are several non-Adafruit boards for CircuitPython, we just changed the "Adafruit Mode" to "CircuitPython Mode" so everyone is equally represented. :-)

**Comment by ShawnHymel** [comment]
Good to know, thanks!

**Pull request merged**